### PR TITLE
feat: add NotImplementedError error for index changing methods Dask

### PR DESCRIPTION
**Pull request by anopsy**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637  
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.
If this looks good I'll also add `NotImplementedError` to the following methods and this PR
`Expr.sort`
`Expr.head`
`Expr.tail`

### CODE REVIEW

File: narwhals/_dask/expr.py [hunk 1]
```diff
@@ -445,6 +450,16 @@ def clip(
             returns_scalar=False,
         )
 
+    def head(self) -> NoReturn:
+        # We can't (yet?) allow methods which modify the index
+        msg = "`Expr.head` is not supported for the Dask backend. Please use `LazyFrame.head` instead."
```
Comment (FBruzzesi) [inline narwhals/_dask/expr.py:455]: Quick heads up: sadly DaskDataframe also doesn't implement `head` (nor `tail`) 🥲

Comment (anopsy) [reply narwhals/_dask/expr.py:455]: 🫠  Where can I check that? I only found Dask API reference for Arrays and DataFrames

Comment (FBruzzesi) [reply narwhals/_dask/expr.py]: `LazyFrame` will point to `DaskLazyFrame` for the dask backend, and the message is suggesting to use such method, so I guess you can check if `narwhals/_dask/dataframe.py`?

Comment (anopsy) [reply narwhals/_dask/expr.py]: image

**Review by MarcoGorelli — APPROVED** [review]
thanks @anopsy ! nice use of noreturn

**Pull request merged**